### PR TITLE
Add tests for CardImage and NotesModal components

### PR DESCRIPTION
**Pull request by karpa4o4**

### CODE REVIEW

**Review by Themanwhosmellslikesugar — APPROVED** [review]

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                    version: 10,
+                    port: "armhf",
+                    packages_url: "https://packages.debian.org/buster/",
+                    build_types: ["classic"],
+                },
+            },
+        },
+    };
+    const image = (status = "Undefined") => ({
+        image_id: "4ed4c369-2fed-412e-b2e1-0d4d176e89fb",
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:38]: Let's use a pattern like `xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx` to make it visible it's fake.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    it("Initial value is substituted in textarea", () => {
+        const initialValue = "Test value";
+        const component = render({ initialValue });
+        expect(component.find("textarea").prop("value")).toEqual(initialValue);
+    });
+
+    /**
+     * Checks input of values in the textarea field.
+     */
+    it("Input of values in textarea", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:87]: Enter value into textarea

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     * Checks if the initial value is substituted in the textarea field.
+     */
+    it("Initial value is substituted in textarea", () => {
+        const initialValue = "Test value";
+        const component = render({ initialValue });
+        expect(component.find("textarea").prop("value")).toEqual(initialValue);
+    });
+
+    /**
+     * Checks input of values in the textarea field.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:85]: Checks entering a value into the textarea.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    it("Succeeded alert isn't displayed", () => {
+        const component = render({});
+        expect(component.find(Alert).prop("show")).toEqual(false);
+    });
+
+    /**
+     * Checks if the initial value is substituted in the textarea field.
+     */
+    it("Initial value is substituted in textarea", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:78]: Specify initial value for textarea

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+     * Checks if the succeeded alert isn't displayed, when the showNotesSucceededMessage
+     * prop is false.
+     */
+    it("Succeeded alert isn't displayed", () => {
+        const component = render({});
+        expect(component.find(Alert).prop("show")).toEqual(false);
+    });
+
+    /**
+     * Checks if the initial value is substituted in the textarea field.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:76]: Checks if the specified initial value is used in the textarea by default.

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            <Provider store={store}>
+                <CardImage
+                    image={image(imageStatus)}
+                />
+            </Provider>,
+        );
+    };
+
+    /**
+     * Checks href and download attributes of download Button.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:68]: Checks the 'href' and 'download' attributes of the download button.

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+        const showedDownloadImageModal = component.find(DownloadInfoModal).find(Modal);
+        const nestedShowedDownloadImageModal = showedDownloadImageModal.children();
+        expect(nestedShowedDownloadImageModal.children().length).toBeGreaterThan(0);
+    });
+
+    /**
+     * Checks if the modal window for adding notes is displayed.
+     */
+    it("Add notes modal is displayed", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:98]: Add notes modal is shown

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+        const showedNotesModal = component.find(NotesModal).find(Modal);
+        const nestedShowedHiddenNotesModal = showedNotesModal.children();
+        expect(nestedShowedHiddenNotesModal.children().length).toBeGreaterThan(0);
+    });
+
+    /**
+     * Checks if the modal window for deleting image is displayed.
+     */
+    it("Delete image modal is displayed", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:116]: Delete image modal is shown

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        const addNotesButton = component.find(Button).at(0);
+        addNotesButton.simulate("click");
+
+        const showedDownloadImageModal = component.find(DownloadInfoModal).find(Modal);
+        const nestedShowedDownloadImageModal = showedDownloadImageModal.children();
+        expect(nestedShowedDownloadImageModal.children().length).toBeGreaterThan(0);
+    });
+
+    /**
+     * Checks if the modal window for adding notes is displayed.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:96]: Checks if the modal window for adding notes is shown.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    it("Modal doesn't rendered", () => {
+        const component = render({ show: false });
+        const modal = component.find(Modal);
+        const nestedModal = modal.children();
+        expect(nestedModal.children().length).toEqual(0);
+    });
+
+    /**
+     * Checks if the succeeded alert is displayed, when the showNotesSucceededMessage
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:58]: Checks if the succeeded alert is shown when the showNotesSucceededMessage prop is true.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+     * prop is true.
+     */
+    it("Succeeded alert is displayed", () => {
+        const component = render({ showNotesSucceededMessage: true });
+        expect(component.find(Alert).prop("show")).toEqual(true);
+    });
+
+    /**
+     * Checks if the succeeded alert isn't displayed, when the showNotesSucceededMessage
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:67]: Checks if the succeeded alert isn't shown when the showNotesSucceededMessage prop is false.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+     * prop is false.
+     */
+    it("Succeeded alert isn't displayed", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:70]: Succeeded alert isn't shown

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+        const modal = component.find(Modal);
+        const nestedModal = modal.children();
+        expect(nestedModal.children().length).toEqual(0);
+    });
+
+    /**
+     * Checks if the succeeded alert is displayed, when the showNotesSucceededMessage
+     * prop is true.
+     */
+    it("Succeeded alert is displayed", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:61]: Succeeded alert is shown

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+                    version: 10,
+                    port: "armhf",
+                    packages_url: "https://packages.debian.org/buster/",
+                    build_types: ["classic"],
+                },
+            },
+        },
+    };
+    const image = (status = "Undefined") => ({
+        image_id: "xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx ",
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:38]: There's a redundant space at the end of the string.

File: src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+     */
+    it("Download Button attributes is valid", () => {
+        const component = render({ imageStatus: "Succeeded" });
+        const downloadButton = component.find(Button).at(0);
+        expect(downloadButton.prop("href")).toBe(`/download-image/${image().image_id}.img.gz`);
+        expect(downloadButton.prop("download")).toBe(true);
+    });
+
+    /**
+     * Checks if the modals window for downloading image shown for image in Undefined status.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx:78]: Checks if the modal window for downloading an image is shown when the status of the image is Undefined.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    it("Modal rendered", () => {
+        const component = render({});
+        expect(component.find(Modal).length).toEqual(1);
+    });
+
+    /**
+     * Checks if the Modal isn't rendered, when the show prop is false.
+     */
+    it("Modal doesn't rendered", () => {
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:50]: Modal isn't rendered

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+    /**
+     * Checks if the Modal is rendered, when the show prop is true.
+     */
+    it("Modal rendered", () => {
+        const component = render({});
+        expect(component.find(Modal).length).toEqual(1);
+    });
+
+    /**
+     * Checks if the Modal isn't rendered, when the show prop is false.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:48]: Checks if Modal isn't rendered when the show prop is false.

File: src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+                    show={show}
+                    imageId={imageId}
+                    initialValue={initialValue}
+                />
+            </Provider>,
+        );
+    };
+
+    /**
+     * Checks if the Modal is rendered, when the show prop is true.
```
Comment (eugulixes) [inline src/modules/Dashboard/components/NotesModal/__tests__/NotesModal.test.jsx:40]: Checks if Modal is rendered when the show prop is true.

**Review by eugulixes — APPROVED** [review]

**Pull request merged**